src/plugins/discover/public/application/main/discover_main_app.tsx

```diff
@@ -100,25 +101,27 @@ export function DiscoverMainApp(props: DiscoverMainProps) {
   useSavedSearchAliasMatchRedirect({ savedSearch, spaces, history });
 
   return (
-    <DiscoverLayoutMemoized
-      dataView={dataView}
-      dataViewList={dataViewList}
-      inspectorAdapters={inspectorAdapters}
-      expandedDoc={expandedDoc}
-      onChangeDataView={onChangeDataView}
-      onUpdateQuery={onUpdateQuery}
-      resetSavedSearch={resetCurrentSavedSearch}
-      setExpandedDoc={setExpandedDoc}
-      navigateTo={navigateTo}
-      savedSearch={savedSearch}
-      savedSearchData$={data$}
-      savedSearchRefetch$={refetch$}
-      searchSource={searchSource}
-      state={state}
-      stateContainer={stateContainer}
-      persistDataView={persistDataView}
-      updateAdHocDataViewId={updateAdHocDataViewId}
-      adHocDataViewList={adHocDataViewList}
-    />
+    <DiscoverAppStateProvider value={stateContainer.appStateContainer}>
+      <DiscoverLayoutMemoized
+        dataView={dataView}
+        dataViewList={dataViewList}
+        inspectorAdapters={inspectorAdapters}
+        expandedDoc={expandedDoc}
+        onChangeDataView={onChangeDataView}
+        onUpdateQuery={onUpdateQuery}
+        resetSavedSearch={resetCurrentSavedSearch}
+        setExpandedDoc={setExpandedDoc}
+        navigateTo={navigateTo}
+        savedSearch={savedSearch}
+        savedSearchData$={data$}
+        savedSearchRefetch$={refetch$}
+        searchSource={searchSource}
+        state={state}
```

Comment:
nit: What about getting rid of each `{state}` prop?

Reply:
thx Dima, yes, I do agree. We will get rid of most of them. So this PR is the start. more will follow